Extensions/3D/JsExtension.js

```diff
@@ -0,0 +1,2393 @@
+        this._defaultDepth = parseFloat(properties.get('depth').getValue());
+        const geometry = new THREE.BoxGeometry(1, 1, 1);
+        const materials = [
+          this._getFaceMaterial(project, materialIndexToFaceIndex[0]),
+          this._getFaceMaterial(project, materialIndexToFaceIndex[1]),
+          this._getFaceMaterial(project, materialIndexToFaceIndex[2]),
+          this._getFaceMaterial(project, materialIndexToFaceIndex[3]),
+          this._getFaceMaterial(project, materialIndexToFaceIndex[4]),
+          this._getFaceMaterial(project, materialIndexToFaceIndex[5]),
+        ];
+        this._threeObject = new THREE.Mesh(geometry, materials);
+        this._threeObject.rotation.order = 'ZYX';
+
+        this._threeGroup.add(this._threeObject);
+      }
+
+      _getFaceMaterial(project, faceIndex) {
+        if (!this._faceVisibilities[faceIndex]) return getTransparentMaterial();
+
+        return this._pixiResourcesLoader.getThreeMaterial(
+          project,
+          this._faceResourceNames[faceIndex],
+          {
+            useTransparentTexture: this._shouldUseTransparentTexture,
+          }
+        );
+      }
+
+      static _getResourceNameToDisplay(objectConfiguration) {
+        return getFirstVisibleFaceResourceName(objectConfiguration);
+      }
+
+      updateThreeObject() {
+        const width = this._instance.hasCustomSize()
+          ? this._instance.getCustomWidth()
+          : this.getDefaultWidth();
+        const height = this._instance.hasCustomSize()
+          ? this._instance.getCustomHeight()
+          : this.getDefaultHeight();
+        let depth;
+        if (this._instance.hasCustomSize()) {
+          const instancePropertyDepth =
+            this._instance.getRawDoubleProperty('depth');
+          depth =
+            typeof instancePropertyDepth === 'number'
+              ? instancePropertyDepth
+              : this._defaultDepth;
+        } else {
+          depth = this._defaultDepth;
+        }
+        const z = this._instance.getRawDoubleProperty('z');
+
+        this._threeObject.position.set(
+          this._instance.getX() + width / 2,
+          this._instance.getY() + height / 2,
+          z + depth / 2
+        );
+
+        this._threeObject.rotation.set(
+          0,
+          0,
+          RenderedInstance.toRad(this._instance.getAngle())
+        );
+
+        if (
+          width !== this._threeObject.scale.width ||
+          height !== this._threeObject.scale.height ||
+          depth !== this._threeObject.scale.depth
+        ) {
+          this._threeObject.scale.set(width, height, depth);
+          this.updateTextureUvMapping();
+        }
+      }
+
+      /**
+       * Updates the UV mapping of the geometry in order to repeat a material
+       * over the different faces of the cube.
+       * The mesh must be configured with a list of materials in order
+       * for the method to work.
+       */
+      updateTextureUvMapping() {
```

Comment:
Unfortunately hard to do because code can't be easily shared between runtime and the editor (and there is also Flow VS TypeScript in addition to the bundling).
Probably we could still go toward this if we change how the runtime is bundled (to support require/import) and how the extension are writtten (TypeScript)